File: src/main/java/fastandslowpointers/linkedlistcycle/CycleDetection.java
package fastandslowpointers.linkedlistcycle;

import utils.LinkedListNode;

public class CycleDetection {

    public boolean detectCycle(LinkedListNode<Integer> head) {
        // Write your code here
        var fast = head;
        var slow = head;

        while(fast != null && fast.next != null) {
            slow = slow.next;
            fast = fast.next.next;
            if(slow == fast) return true;

        }
        return false;

    }

}

File: src/main/kotlin/fastandslowpointers/linkedlistcycle/CycleDetection.kt
package fastandslowpointers.linkedlistcycle

import utilskotlin.LinkedListNode

fun detectCycle(head: LinkedListNode<Int>): Boolean {

    var slow = head
    var fast = head

    while (true) {
        slow = slow.next!!
        fast = fast.next?.next?:break
        if (slow == fast) return true
    }
    return false
}
